samples/account-linking/csharp/Source/Services/AccountLinkingStateService/OAuthStateServiceOptions.cs

```diff
@@ -0,0 +1,29 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Teams.Samples.AccountLinking.State;
+
+/// <summary>
+/// The configuration options for the <see cref="AccountLinkingStateService" />
+/// </summary>
+public sealed class AccountLinkingStateServiceOptions
+{
+
+    /// <summary>
+    /// Gets or sets how long the state tokens should be valid for.
+    /// </summary>
+    /// <returns>The timespan of how long tokens are valid.</returns>
+    /// <remarks>
+    /// In practice this is a balancing act between how long we want to allow users to take to log in and how long we wish to 
+    /// store token ids for detecting replays.
+    /// </remarks>
+    [Range(typeof(TimeSpan), "00:00:30", "10675199.02:48:05.4775807")]
```

Comment:
But what if I need it to expire in 30,000 years :(

Reply:
The upper bound is a tad silly, just what the value was returned as for `TimeSpan.Max`.